### DISCUSSION TITLE
Behaviour of bundled recv: will kernel post a single CQE only after reading all the requested bytes?

### DISCUSSION THREAD

**Discussion by ghost** [axboe/liburing#1167] [Q&A] [answered]
Hello,

https://git.kernel.dk/cgit/linux/commit/?h=io_uring-recvsend-bundle&id=be6c58d858be3d41e33b98a46bd1ce2e3e7d4e92

The commit message above suggests that when using bundled send, a single CQE will be posted if either kernel runs out of buffers to send or requested bytes were processed.

I should probably just study the diff but I was wondering if bundled recv will also only post a single CQE only after completely reading all the requested bytes (provided that there are enough buffers available) or not.

Thanks,

**Comment by axboe** [comment, marked as answer]
For recv, it'll post a CQE for the data that is available, as a single CQE. Bundled send and receive behave in the same way in that regard.
